### ISSUE TITLE
'utils. deepMerge()' converts an array to an object

### ISSUE THREAD

**Issue by Alanscut** [axios/axios#2567]
### Describe the issue
When I read the 'Axios' source code, I found that' utils. deepMerge() 'will convert arrays to objects, which is unreasonable.
https://github.com/axios/axios/blob/master/lib/utils.js#L286
### Code
```
var deepMerge = require('./lib/utils').deepMerge;
var a = {foo: {bar: 111}, hnd: [444, 555, 666]};
var b = {foo: {baz: 456}, hnd: [456, 789]};
console.log(deepMerge(a, b));
```
### result
```
{ foo: { bar: 111, baz: 456 }, hnd: { '0': 456, '1': 789, '2': 666 } }
```

### Expected behavior
Executing 'utils.deepMerge()' does not change the data type.
```
{ foo: { bar: 111, baz: 456 }, hnd: [ 456, 789, 666 ] }
```

### Environment:
Axios Version 0.19.0

**Comment by yaozaiyu** [comment]
请问这个问题如何解决？没法看你改的内容

**Comment by Alanscut** [comment]
Please see https://github.com/axios/axios/pull/2568/files#

**Comment by chinesedfan** [comment]
Closed due to #2844.
